src/main/java/seedu/address/ui/PaymentWindow.java

```diff
@@ -0,0 +1,92 @@
+package seedu.address.ui;
+
+import java.io.IOException;
+import java.util.logging.Logger;
+
+import com.google.zxing.WriterException;
+
+import javafx.fxml.FXML;
+import javafx.scene.control.Button;
+import javafx.scene.control.Tooltip;
+import javafx.scene.image.Image;
+import javafx.scene.image.ImageView;
+import javafx.stage.Stage;
+import seedu.address.commons.core.LogsCenter;
+import seedu.address.logic.paynow.PayNowCode;
+import seedu.address.model.person.Person;
+
+/**
+ * Controller for the Payment window
+ */
+public class PaymentWindow extends UiPart<Stage> {
+    private static final String FXML = "PaymentWindow.fxml";
+    private static final Logger logger = LogsCenter.getLogger(PaymentWindow.class);
+    private static final String HELP_TOOLTIP_TEXT = "Open your banking application and scan this QR code to pay %s!\n"
+            + "Note that this QR code only works if the mobile number is registered to a bank account.";
+    private final Person personToPay;
+    private final Runnable onResetDebt;
+
+    @FXML
+    private ImageView qrCode;
+    @FXML
+    private Button cancelButton;
+
+    @FXML
+    private Button resetButton;
+    @FXML
+    private Tooltip helpTooltip;
+
+    /**
+     * Creates a new QrWindow.
+     *
+     * @param person The person whom the user is attempting to pay.
+     */
+    public PaymentWindow(Person person, Runnable onResetDebt) throws IOException, WriterException {
+        super(FXML, new Stage());
+        this.personToPay = person;
+        this.onResetDebt = onResetDebt;
+        Image image = new Image(PayNowCode.generatePayNowQrCode(
+                person.getPhone().value, Math.max(0, -person.getMoneyOwed().moneyOwed)));
```

Comment:
Could maybe use the getAmount method in MoneyOwed instead of accessing the attribute

Reply:
Will add that in after you merge your fork into our master :)